doc-src/ch-run_desc.rst

```diff
@@ -184,6 +207,127 @@ are tweaked. This section lists those changes.
     * `Fedora <https://fedoraproject.org/wiki/Features/UsrMove>`_
     * `Debian <https://wiki.debian.org/UsrMerge>`_
 
+Setting environment variables with :code:`--set-env-file`
+---------------------------------------------------------
+
+The purpose of :code:`--set-env-file[=FILE]` is to set environment variables
+originating by mechanisms such as Dockerfile :code:`ENV` statements or the
+Bash :code:`set` command.
+
+:code:`FILE` is a guest path. If not specified, it defaults to **FIXME**.
+
+The syntax of :code:`FILE` is key-value pairs separated by the first equals
+character (:code:`=`, ASCII 61), with optional single straight quotes
+(:code:`'`, ASCII 39) around the value stripped. No variable expansion,
+comments, etc. are provided. The value may be empty, but not the key. (This
+syntax is designed to accept the output of **FIXME: DOCKER SOMETHING** and
+shell :code:`set` or :code:`env`.) Examples of valid lines:
+
+.. list-table::
+   :header-rows: 1
+
+   * - Line
+     - Key
+     - Value
+   * - :code:`FOO=bar`
+     - :code:`FOO`
+     - :code:`bar`
+   * - :code:`FLAGS=-march=foo -mtune=bar`
+     - :code:`FLAGS`
+     - :code:`-march=foo -mtune=bar`
+   * - :code:`FLAGS='-march=foo -mtune=bar'`
+     - :code:`FLAGS`
+     - :code:`-march=foo -mtune=bar`
+   * - :code:`FOO=`
+     - :code:`FOO`
+     - (empty string)
+   * - :code:`FOO=''`
+     - :code:`FOO`
+     - (empty string)
+   * - :code:`FOO=''''`
+     - :code:`FOO`
+     - :code:`''` (two single quotes)
+
+Example invalid lines:
+
+.. list-table::
+   :header-rows: 1
+
+   * - Line
+     - Problem
+   * - :code:`FOO bar`
+     - no separator
+   * - :code:`=bar`
+     - key cannot be empty
+
+Example valid lines that are probably not what you want:
+
+.. Note: Plain leading space screws up ReST parser. We use ZERO WIDTH SPACE
+   U+200B, then plain space. This will copy and paste incorrectly, but that
+   seems unlikely.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Line
+     - Key
+     - Value
+     - Problem
+   * - :code:`FOO="bar"`
+     - :code:`FOO`
+     - :code:`"bar"`
+     - double quotes aren't stripped
+   * - :code:`FOO=bar # baz`
+     - :code:`FOO`
+     - :code:`bar # baz`
+     - comments not supported
+   * - :code:`PATH=$PATH:/opt/bin`
+     - :code:`PATH`
+     - :code:`$PATH:/opt/bin`
+     - variables not expanded
+   * - :code:`​ FOO=bar`
+     - :code:`​ FOO`
+     - :code:`bar`
+     - leading space in key
+
+The variables in :code:`FILE` replace any already set.
```

Comment:
I much prefer variables in the FILE replace any already set. However, I seem to recall you saying that they should not. Has your opinion changed or am I recalling incorrectly?

Reply:
I don't remember. :)